src/movement/test/test_thruster_maestro.cpp

```diff
@@ -0,0 +1,146 @@
+#include "utility/test_tools.hpp"
+#include <vector>
+#include <string>
+
+rs::Serial mSerial;
+ros::Publisher pub;
+double byte_check(uint8_t byte2, uint8_t byte3);
+std::vector<uint8_t> channels;
+double max_speed = 0.6;
+
+//insert tests here
+TEST(Maestro, basicTest)
+{
+    //allocate an array of bytes that will be used to receive serial data from
+    //the serial port. I use the arbitrarily large size of
+    uint8_t maestro_data[256] = {0};
+
+
+    //create an empty thruster message to send to the thruster module
+    robosub::thruster maestro_msg;
+    for (uint8_t i = 0; i < channels.size(); i++)
+    {
+        maestro_msg.data.push_back(0.0);
+    }
+
+    //send initial message to make the thruster module send it's reset signal.
+    //This test isn't checking for it, so we just flush the serial buffer and
+    //throw it away
+    pub.publish(maestro_msg);
+    ros::Duration(2).sleep();
+    mSerial.Flush();
+
+    //fill out data in the thruster message.
+    maestro_msg.data.clear();
+    maestro_msg.data.push_back(1.0);
+    maestro_msg.data.push_back(0.7);
+    maestro_msg.data.push_back(0.8);
+    maestro_msg.data.push_back(-0.5);
+    maestro_msg.data.push_back(-0.8);
+    maestro_msg.data.push_back(-1.0);
+
+    //publish the thruster message
+    pub.publish(maestro_msg);
+    ros::Duration(1).sleep();
+
+    //after we published the thruster message, the thruster module should have
+    //received the message, and then sent some data down the serial port, here
+    //we read that data and verify it is correct
+    for(unsigned int i = 0; i < maestro_msg.data.size(); ++i)
+    {
+        //read one byte from the serial port
+        int bytes_rxd = mSerial.Read(maestro_data, 4);
+        ROS_INFO("recieved %d bytes", bytes_rxd);
+        //mSerial.Read(maestro_data, 4);
+
+        ROS_WARN("raw_vals: %x, %x", maestro_data[2], maestro_data[3]);
+        double actual_speed = byte_check(maestro_data[2], maestro_data[3]);
+
+
+        //checks to make sure the first byte in maestro_data is correct
+        EXPECT_EQ(maestro_data[0], 0x84);
+        EXPECT_EQ(maestro_data[1], channels[i]);
+
+        //compare and make sure the value recieved is equal to the value sent
+        double expected_speed = maestro_msg.data[i];
+        if(fabs(expected_speed) > max_speed)
+        {
+            expected_speed = max_speed * ((expected_speed < 0)? -1 : 1);
+        }
+        EXPECT_FLOAT_EQ(actual_speed, expected_speed);
+    }
+
+    ROS_INFO("test over, waiting");
+}
+
+double byte_check(uint8_t byte2, uint8_t byte3)
+{
+  double target;
+  uint16_t pulse_width;
+
+  pulse_width = ((byte3 << 7) | byte2) >> 2;
+  target = static_cast<double>(pulse_width - 1500) / 400;
+
+  return target;
+}
+
+
+int main(int argc, char *argv[])
+{
+    //initialize the google testing framework
+    testing::InitGoogleTest(&argc, argv);
+
+    //initialize our ros node
+    ros::init(argc, argv, "maestro_launch_test");
```

Comment:
As long as the name is unique there really isn't a need to change the running node name here.